### PR TITLE
Refactor RSA PSS MGF processing to enable future use by OAEP.

### PR DESCRIPTION
**Pull request by briansmith**
Previously, `mgf1()` wrote the mask to the buffer, and then we XOR'd
the data onto the mask. Now, `mgf1()` XOR's the mask onto the data
that is already in the `out` buffer.

### CODE REVIEW

**Comment by briansmith** [comment]
@complexspaces PTAL.

File: src/rsa/padding.rs [hunk 1]
```diff
@@ -269,42 +265,36 @@ impl RsaEncoding for PSS {
...
-        // Step 4.
-        let mut salt = [0u8; MAX_SALT_LEN];
-        let salt = &mut salt[..metrics.s_len];
-        rng.fill(salt)?;
-
-        // Step 5 and 6.
-        let h_hash = pss_digest(self.digest_alg, m_hash, salt);
+        let (db, digest_terminator) = em.split_at_mut(metrics.db_len);
+        let h;
+        {
```
Comment (complexspaces) [inline src/rsa/padding.rs:270]: Nit: `h` could be assigned to the output of this block to potentially clarify at what point its bound when scrolling the code.

Comment (briansmith) [reply src/rsa/padding.rs]: Good point. I actually just eliminated the scoping block as it's no longer important with the refactored code.

**Review by complexspaces — APPROVED** [review]

File: src/rsa/padding.rs (hunk 1 above)
Comment (complexspaces) [reply src/rsa/padding.rs]: That works too. I think its much easier to follow whats happening now 👍

**Pull request merged**